### PR TITLE
Polyfill `instancesync` & `update-server` for ci

### PR DESCRIPTION
**Pull request by Quezler**
After taking a closer look at what goes wrong if you don't call `update-server.ps1` it basically boils down to just `settings.cfg` and `start-automated-server.sh` not being able to find one another:
```
> Run node .github/actions/ci/start-server.js bash ./automation/start-automated-server.sh
./automation/start-automated-server.sh: line 161: settings.cfg: No such file or directory
```
Apart from copying those files the entire rest of the file seems to only do stuff related to detecting & pulling changes from a remote git repository, but during ci the changes are already pulled/checked-out, so we can bypass it just fine.

Since `update-server.ps1` pulled it needed the `instancesync setup` to hook up some git events, but we don't need that anymore since we're only pulling once + a manual `instancesync` afterwards, so that has been simplified as well.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,10 +31,13 @@ jobs:
                   node .github/actions/ci/packmode.js ${{ matrix.packmode }} > mode.json
                   cat mode.json
 
-            - run: bash ./InstanceSyncSetup.sh
-              working-directory: ./automation
-            - run: pwsh ./update-server.ps1 ${{ github.ref_name }} world backups $true $false
-              working-directory: ./automation
+            - name: cd automation && bash ./InstanceSyncSetup.sh
```
Comment (NielsPilgaard) [inline .github/workflows/ci.yml]: The name is a bit missleading I think, how about `Download Mods` :D

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -31,10 +31,13 @@ jobs:
                   node .github/actions/ci/packmode.js ${{ matrix.packmode }} > mode.json
                   cat mode.json
 
-            - run: bash ./InstanceSyncSetup.sh
-              working-directory: ./automation
-            - run: pwsh ./update-server.ps1 ${{ github.ref_name }} world backups $true $false
-              working-directory: ./automation
+            - name: cd automation && bash ./InstanceSyncSetup.sh
+              run: java -jar InstanceSync.jar
+            - name: cd automation && pwsh ./update-server.ps1
```
Comment (NielsPilgaard) [inline .github/workflows/ci.yml]: Name suggestion: `Start Minecraft Server`

**Review by NielsPilgaard — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Quezler) [reply .github/workflows/ci.yml]: Well i was trying to keep the referring to the equivalent of the terminology of the server upgrade guide (aka: which step it mimicked), but i'll update it if you'd like 🤔

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -31,10 +31,13 @@ jobs:
...
-        working-directory: ./automation
-      - run: pwsh ./update-server.ps1 ${{ github.ref_name }} world backups $true $false
-        working-directory: ./automation
+      - name: cd automation && bash ./InstanceSyncSetup.sh
+        run: java -jar InstanceSync.jar
+      - name: cd automation && pwsh ./update-server.ps1
+        run: |
+          cp ./automation/settings.cfg ./settings.cfg
+          cp ./automation/start-automated-server.sh ./start-automated-server.sh
+          pwsh ./automation/remove-client-mods.ps1
```
Comment (Quezler) [inline .github/workflows/ci.yml:40]: if you really insist on renaming them i would suggest accepting this one: (edit access for maintainers is on)
```suggestion
      - name: Install server
        run: |
          java -jar InstanceSync.jar
          cp ./automation/settings.cfg ./settings.cfg
          cp ./automation/start-automated-server.sh ./start-automated-server.sh
          pwsh ./automation/remove-client-mods.ps1
```

**Review by NielsPilgaard — APPROVED** [review]

**Pull request merged**